controllers/workspace/devworkspace_controller.go

```diff
@@ -170,13 +157,11 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 	}
 	flattenedWorkspace, err := flatten.ResolveDevWorkspace(workspace.Spec.Template, flattenHelpers)
 	if err != nil {
-		reqLogger.Info("DevWorkspace start failed")
-		reconcileStatus.Phase = devworkspace.WorkspaceStatusFailed
-		// TODO: Handle error more elegantly
-		reconcileStatus.Conditions[devworkspace.WorkspaceFailedStart] = fmt.Sprintf("Error processing devfile: %s", err)
-		return reconcile.Result{}, nil
+		return r.failWorkspace(fmt.Sprintf("Error processing devfile: %s", err), reqLogger, &reconcileStatus)
```

Comment:
Does it make any sense if we duplicate message to condition where it happened?
```suggestion
		reconcileStatus.setConditionFalse(ResolvedDevWorkspace, fmt.Sprintf("Error processing devfile: %s", err))
		return r.failWorkspace(fmt.Sprintf("Error processing devfile: %s", err), reqLogger, &reconcileStatus)
```

Reply:
or simpler way, if failWorkspace would supply the condition.
```suggestion
		return r.failWorkspace(ResolvedDevWorkspace, fmt.Sprintf("Error processing devfile: %s", err), reqLogger, &reconcileStatus)
```

Reply:
I thought about this a bit more and I'm not sure it makes total sense:
- If we want to move setting the condition into failWorkspace, it makes failWorkspace depend on a condition, which we don't have for e.g. when `validateCreatorLabel()` fails. We could add a condition there as well (`WorkspaceInitialized` perhaps), but I also would prefer to set conditions when they're useful (`WorkspaceInitialized` will be true in 99% of cases, and when it's not the `FailedStart` condition will contain the same info).
- Further, failure points in the reconcile loop aren't directly tied to conditions. For example, we could fail parsing resourceLimits, which happens after we set `resolveDevWorkspace` but before `storageProvisioned`. I'd fear it would require us to add too many conditions to represent rare edge cases.

Reply:
The issue I had in mind, if phase throws a common error, it may help to understand on which condition is happened, like ResolveDevWorkspace has error `could not unmarshal devfile from response: ${err}` where error is like, is not a valid yaml.
^ but that error can be improved as well, to contain the related component failure, or something. Maybe it's already done on some step of throwing error up, but I assume we just propagate err, without provisioning details.

I agree that in some cases it may be a bit redundant, like `failed to resolve devworkspace: parent has invalid URL`

About cases, where error happens independent from phase, the phase may be optional.

So, I agree that my suggestion still has some question, but I agree to proceed without it.

Reply:
Well, the user-facing messages should be good enough, since you solve the issue with face identifying by appending prefix to error message.